### PR TITLE
[popover] Don't stick upon click with `openOnHover` prop

### PR DESCRIPTION
**Pull request by atomiks**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

The upgrade fixes bugs and adds this prop to turn off.

### CODE REVIEW

**Comment by michaldudak** [comment]
Seems to work fine.

> adds this prop to turn off.

There's no prop exposed to control this behavior?

**Comment by atomiks** [comment]
> There's no prop exposed to control this behavior?

Not currently but there could be. @vladmoroz

**Comment by vladmoroz** [comment]
I wouldn't want to add a prop unless necessary—is there a typical design where it makes sense to keep it open on click?

**Comment by atomiks** [comment]
> I wouldn't want to add a prop unless necessary—is there a typical design where it makes sense to keep it open on click?

Back when I was developing tippy.js, the "stick" behavior was actually the requested default (a while ago, 2019), and it wasn't configurable. I didn't hear of complaints. It's also the default in Floating UI now (working better with the recent fixes).

File: packages/react/src/Popover/Trigger/PopoverTrigger.test.tsx [hunk 1]
```diff
@@ -51,9 +51,29 @@ describe('<Popover.Trigger />', () => {
...
+      );
+
+      const trigger = screen.getByRole('button');
+
+      await user.hover(trigger);
+
+      // Closes instantly after opening due to `useClick` prop `stickIfOpen=false`
+      await act(async () => {
+        trigger.click();
+      });
```
Comment (atomiks) [inline packages/react/src/Popover/Trigger/PopoverTrigger.test.tsx:68]: Perhaps this is why it's configured by default, but it depends on the `delay` you use. If you happen to click <50ms after it opens coincidentally, it's pretty unexpected

**Review by vladmoroz — APPROVED** [review]

**Pull request merged**